main.go

```diff
@@ -133,6 +137,46 @@ func (d *Downloader) downloadFileWithTimeout(userCtx context.Context, u string)
 	}
 }
 
+func (d *Downloader) getSize(ctx context.Context, u string) (uint64, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodHead, u, nil)
+	if err != nil {
+		return 0, fmt.Errorf("error creating the request for %s: %w", u, err)
+	}
+	resp, err := d.client.Do(req)
+	if err != nil {
+		return 0, fmt.Errorf("error sending a get http request to %s: %w", u, err)
+	}
+	if resp.StatusCode != 200 {
+		return 0, fmt.Errorf("status code is not valid: %v", resp.StatusCode)
+	}
+	defer resp.Body.Close()
+	if resp.ContentLength != 0 && resp.ContentLength != -1 {
```

Comment:
Wouldn't Content-Range check be the priority? I mean, first, check it, then the other case?

Reply:
That's a good question, I don't know the answer to this question actually. If we take this [https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests#:~:text=The%20Content%2DLength%20header%20now,resource%20this%20partial%20message%20belongs.](Mozilla specification) as true:

```
The Content-Length header now indicates the size of the requested range (and not the full size of the image). The Content-Range response header indicates where in the full resource this partial message belongs.
```

I would use the Content-Length header first.

Reply:
> I would use the Content-Length header first.

I wouldn't. I would check the HTTP status first.
If it's not a `200 OK` we might be reading the wrong kind of response.

Reply:
I think we have two statuses we must read from 200 and 206 (OK and Partial Content).

Reply:
Do you have a scenario in which a `HEAD` request returns partial content?

Reply:
Sounds like a 200-only check. https://stackoverflow.com/questions/57909738/download-doesnt-work-when-using-code-206-partial-content

Reply:
Yes.

A `HEAD` request means a request only for the headers (no body in the request).

A 206 response means that the body in the request is a part of the full body.

Why are we checking the status of a partial body for a request we expect no body at all?

Reply:
So should we get rid of the status code part?

Reply:
[Nope](https://github.com/cuducos/chunk/pull/10#discussion_r1021976330).

Reply:
When I say:

> If it's not a `200 OK` we might be reading the wrong kind of response.

I think of something like:

```go
if resp.StatusCode != 200 {
    return 0, fmt.Errorf("got unexpected http response status for %s: %s", u, resp.Status)
}

Reply:
Awesome! Roger that =)!